src/commands/deploy/genezio.ts

```diff
@@ -230,6 +234,55 @@ export async function genezioDeploy(options: GenezioDeployOptions) {
         });
     }
 
+    if (configuration.services?.crons) {
+        const res = await getFunctions(deployClassesResult?.projectEnvId || "").catch(() => {
+            throw new UserError("Something went wrong while fetching the cron functions.");
+        });
+        const crons: CronDetails[] = [];
+        for (const cron of configuration.services.crons) {
+            let cronFound = false;
+            const yamlFunctionName = await evaluateResource(
+                configuration,
+                cron.function,
+                options.stage || "prod",
+                undefined,
+                undefined,
+            );
+            for (const deployedFunction of res.functions) {
+                if (deployedFunction.name === `function-${yamlFunctionName}`) {
+                    crons.push({
+                        name: cron.name,
+                        url: deployedFunction.cloudUrl,
+                        endpoint: cron.endpoint || "",
+                        cronString: cron.schedule,
+                    });
+                    cronFound = true;
+                    break;
+                }
+            }
+            if (!cronFound) {
+                throw new UserError(
+                    `Function ${yamlFunctionName} not found in the deployed functions. Please make sure the function is deployed before adding it to the cron.`,
+                );
+            }
```

Comment:
This is too much logic here to prevent registering a cron without a function.

It would be better to attempt to register the cron and let the backend fail if there is no function deployed.

This will simplify a lot this code and improve performance (no need to fetch all the functions, iterate through them etc.)

Please rewrite this section with this philosophy in mind.